### ISSUE TITLE
Make use of same socket expose for both podman and docker client

### ISSUE THREAD

**Issue by praveenkumar** [code-ready/crc#3020]
As of now we do something following but if we are exposing the socket then we might just use it for both podman/docker client instead using ssh implementation for podman and socket expose for docker.

```
$ ./crc podman-env
export PATH="/Users/prkumar/.crc/bin/oc:$PATH"
export CONTAINER_SSHKEY="/Users/prkumar/.crc/machines/crc/id_ecdsa"
export CONTAINER_HOST="ssh://core@127.0.0.1:2222/run/user/1000/podman/podman.sock"
export DOCKER_HOST="unix:///Users/prkumar/.crc/machines/crc/docker.sock"
```

```
$ podman --url unix:///Users/prkumar/.crc/machines/crc/docker.sock images --log-level debug
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called images.PersistentPreRunE(podman --url unix:///Users/prkumar/.crc/machines/crc/docker.sock images --log-level debug) 
DEBU[0000] DoRequest Method: GET URI: http://d/v3.4.4/libpod/_ping 
DEBU[0000] DoRequest Method: GET URI: http://d/v3.4.4/libpod/images/json 
REPOSITORY                       TAG                                       IMAGE ID      CREATED       SIZE
quay.io/crcont/gvisor-tap-vsock  3231aba53905468c22e394493a0debc1a6cc6392  f42b05e6b25e  5 months ago  6.65 MB
DEBU[0000] Called images.PersistentPostRunE(podman --url unix:///Users/prkumar/.crc/machines/crc/docker.sock images --log-level debug) 
```

We can have something like and expose both root/rootless socket.
```
$ ./crc podman-env
export PATH="/Users/prkumar/.crc/bin/oc:$PATH"
export CONTAINER_HOST="unix:///Users/prkumar/.crc/machines/crc/docker-rootless.sock"
export DOCKER_HOST="unix:///Users/prkumar/.crc/machines/crc/docker-rootless.sock"
```

```
$ ./crc podman-env --root
export PATH="/Users/prkumar/.crc/bin/oc:$PATH"
export CONTAINER_HOST="unix:///Users/prkumar/.crc/machines/crc/docker.sock"
export DOCKER_HOST="unix:///Users/prkumar/.crc/machines/crc/docker.sock"
```

**Comment by cfergeau** [comment]
If we go that way, I'd rename `docker.sock` to `podman.sock` ;) Keeping the current way, or using the unix socket directly are roughly equivalent, both will involve a ssh tunnel.

Even if we don't do that, renaming the socket would be a good thing, as it is not only a 'docker compatibility socket', it can also be used to interact with the podman-only REST API over curl or any other tool. I'd favour `podman.sock` so that it's sligthly clearer that this socket is not restricted to `docker` things.

**Comment by benoitf** [comment]
Yes, I think naming the socket `podman.sock` instead of `docker.sock` is better.
Why: because people may expect the same behaviour than with podman machine v4.0.0

There is only websocket mounted and the name is `podman.sock`

When creating the machine, default is rootless, so the `podman.sock` file is the rootless socket

If we init the machine using `--rootful` parameter then the `podman.sock` is the rootful socket

And there is an optional tool called `podman-mac-helper` / https://github.com/containers/podman/tree/main/cmd/podman-mac-helper that is trying to see if there is something like docker desktop and if not, it will sync `/var/run/docker.sock` from the mounted local `podman.sock` file

**Comment by cfergeau** [comment]
This has been implemented in 790195bb3baafbb2 but using docker.sock as the name. Keeping this open to remember to change it.
